api/envoy/config/listener/v3/listener_components.proto

```diff
@@ -206,6 +207,17 @@ message FilterChain {
   // name is provided, Envoy will allocate an internal UUID for the filter chain. If the filter
   // chain is to be dynamically updated or removed via FCDS a unique name must be provided.
   string name = 7;
+
+  // [#not-implemented-hide:] Whether the filter chain will be built on demand or not.
```

Comment:
I would recommend factoring this into an `OnDemandConfiguration` message or similar, and then having `rebuild_timeout` but a field in that message. The presence of an empty `on_demand_configuration` would turn on on-demand behavior. This will allow us to add additional configuration options in the future (like potentially a default filter chain if the on-demand build fails, or a default filter chain that is used while the on-demand fetch is performed.)

Reply:
Great idea!